contracts/staking/StakeManager.sol

```diff
@@ -139,7 +143,13 @@ contract StakeManager is Validator, IStakeManager, RootChainable, Lockable {
 
   function startAuction(uint256 validatorId, uint256 amount) external {
     require(isValidator(validatorId));
+    // when dynasty period is updated validators are in cool down period
+    require(replacementCoolDown == 0 || replacementCoolDown <= currentEpoch, "Cool down period");
     require(auctionPeriod >= currentEpoch.sub(validatorAuction[validatorId].startEpoch), "Invalid auction period");
+    // dynasty--auctionPeriod--dynasty--auctionPeriod--dynasty
+    // make sure that its `auctionPeriod` window
+    require((currentEpoch.sub(validators[validatorId].activationEpoch) % dynasty.add(auctionPeriod)) > dynasty, "Not an auction time");
```

Comment:
A comment saying what this formula means might be a good idea.

Reply:
yeah makes sense will add.

Reply:
i am sayli